lib/view/item_widget.dart

```diff
@@ -39,37 +64,57 @@ class _ItemWidgetState extends State<ItemWidget> {
           child: Checkbox(
             value: widget.item.isChecked,
             onChanged: _onNewCheckedState,
-            visualDensity: const VisualDensity(
-                horizontal: VisualDensity.maximumDensity,
-                vertical: VisualDensity.maximumDensity),
+            visualDensity: VisualDensity.compact,
           ));
 
       textDecoration =
           widget.item.isChecked ? TextDecoration.lineThrough : null;
     }
 
-    return InkWell(
-        onTap: widget.tappable ? _showEditDialog : null,
-        child: Padding(
+    Widget textField = TextField(
+      controller: _controller,
+      decoration:
+          const InputDecoration(border: InputBorder.none, isDense: true),
+      style: itemTextStyle.copyWith(decoration: textDecoration),
+      focusNode: _focusNode,
+      enabled: !widget.item.isChecked,
+    );
+
+    return GestureDetector(
+        child: Container(
+            // For some reason this makes the gesture detector work
+            color: Colors.transparent,
```

Comment:
Why do we need the gesture detector when the focus node (of the text field) already calls `widget.onFocus()` in its listener? I tried removing it, but I couldn't see any difference in functionality.

Reply:
you should be able to tap on ANY part of the item to select it, not just the text field.